Compiler warning, deprecated header used by HPX itself

## Expected Behavior

Compile without warning about use of deprecated header.

## Actual Behavior

Compiler warns about use of deprecated header. Minor issue I guess. Reporting anyway.

```
/tmp/hello_hpx-fetch_content/RelWithDebInfo-apex/build/_deps/hpx-src/libs/lcos_local/include_compatibility/hpx/local_lcos/composable_guard.hpp:19:2: warning: #warning "The header hpx/local_lcos/composable_guard.hpp is deprecated,     please include hpx/lcos_local/composable_guard.hpp instead"
[-Wcpp]
/tmp/hello_hpx-fetch_content/RelWithDebInfo-apex/build/_deps/hpx-src/libs/runtime_local/include_compatibility/hpx/util/interval_timer.hpp:18:2: warning: #warning "The header hpx/util/interval_timer.hpp is deprecated,     please include hpx/modules/runtime_local.hpp instead" [-Wcpp]
```

It seems compatibility headers are including each other.

## Steps to Reproduce the Problem

I used this configuration:
```
cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DHELLO_HPX_REPOSITORY_CACHE:PATH=/home/kor/development/repository -DHPX_WITH_APEX:BOOL=ON -DAPEX_WITH_OTF2:BOOL=ON -DHELLO_HPX_BUILD_OTF2:BOOL=ON -DHPX_WITH_EXAMPLES:BOOL=ON -DHPX_WITH_MALLOC:STRING=tcmalloc -DHPX_WITH_HWLOC:BOOL=ON -S /home/kor/development/project/hello_hpx -B /tmp/hello_hpx-fetch_content/RelWithDebInfo-apex/build`
cmake --build /tmp/hello_hpx-fetch_content/RelWithDebInfo-apex/build
```
(See also: https://github.com/kordejong/hello_hpx)

## Specifications

  - HPX Version: 1.5.0-rc1
  - Platform (compiler, OS): Linux, gcc-7

--- a/hpx/lcos/detail/promise_base.hpp
+++ b/hpx/lcos/detail/promise_base.hpp
@@ -94,7 +94,7 @@ namespace lcos {
               : alloc_(alloc)
             {}
 
-            promise_data_allocator(init_no_addref no_addref, in_place in_place,
+            promise_data_allocator(init_no_addref no_addref, in_place,
                     other_allocator const& alloc)
               : promise_data<Result>(no_addref), alloc_(alloc)
             {}

--- a/libs/actions_base/include/hpx/actions_base/basic_action.hpp
+++ b/libs/actions_base/include/hpx/actions_base/basic_action.hpp
@@ -348,7 +348,7 @@ namespace hpx { namespace actions {
         ///////////////////////////////////////////////////////////////////////
         template <typename IdOrPolicy, typename Policy, typename... Ts>
         HPX_FORCEINLINE static result_type sync_invoke(Policy const& policy,
-            IdOrPolicy const& id_or_policy, error_code& ec, Ts&&... vs)
+            IdOrPolicy const& id_or_policy, error_code&, Ts&&... vs)
         {
             return hpx::sync<basic_action>(
                 policy, id_or_policy, std::forward<Ts>(vs)...);

--- a/libs/actions_base/include/hpx/actions_base/plain_action.hpp
+++ b/libs/actions_base/include/hpx/actions_base/plain_action.hpp
@@ -79,7 +79,7 @@ namespace hpx { namespace actions {
 
         template <typename... Ts>
         static R invoke(naming::address::address_type /*lva*/,
-            naming::address::component_type comptype, Ts&&... vs)
+            naming::address::component_type /*comptype*/, Ts&&... vs)
         {
             basic_action<detail::plain_function, R(Ps...),
                 derived_type>::increment_invocation_count();

--- a/libs/async_distributed/include/hpx/async_distributed/detail/sync_implementations.hpp
+++ b/libs/async_distributed/include/hpx/async_distributed/detail/sync_implementations.hpp
@@ -30,7 +30,7 @@ namespace hpx { namespace detail {
     {
         template <typename... Ts>
         HPX_FORCEINLINE static Result call(
-            naming::id_type const& id, naming::address&& addr, Ts&&... vs)
+            naming::id_type const& /*id*/, naming::address&& addr, Ts&&... vs)
         {
             typedef typename Action::remote_result_type remote_result_type;
 
@@ -47,7 +47,7 @@ namespace hpx { namespace detail {
     {
         template <typename... Ts>
         HPX_FORCEINLINE static void call(
-            naming::id_type const& id, naming::address&& addr, Ts&&... vs)
+            naming::id_type const& /*id*/, naming::address&& addr, Ts&&... vs)
         {
             Action::execute_function(
                 addr.address_, addr.type_, std::forward<Ts>(vs)...);

--- a/libs/components_base/include/hpx/components_base/traits/component_pin_support.hpp
+++ b/libs/components_base/include/hpx/components_base/traits/component_pin_support.hpp
@@ -19,7 +19,7 @@ namespace hpx { namespace traits {
         struct pin_helper
         {
             template <typename Component>
-            static constexpr void call(wrap_int, Component* p)
+            static constexpr void call(wrap_int, Component*)
             {
             }
 
@@ -34,7 +34,7 @@ namespace hpx { namespace traits {
         struct unpin_helper
         {
             template <typename Component>
-            static constexpr bool call(wrap_int, Component* p)
+            static constexpr bool call(wrap_int, Component*)
             {
                 return false;
             }
@@ -51,7 +51,7 @@ namespace hpx { namespace traits {
         struct pin_count_helper
         {
             template <typename Component>
-            static constexpr std::uint32_t call(wrap_int, Component* p)
+            static constexpr std::uint32_t call(wrap_int, Component*)
             {
                 return 0;
             }

--- a/libs/coroutines/include/hpx/coroutines/detail/coroutine_stackless_self.hpp
+++ b/libs/coroutines/include/hpx/coroutines/detail/coroutine_stackless_self.hpp
@@ -34,7 +34,7 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail {
         {
         }
 
-        arg_type yield_impl(result_type arg) override
+        arg_type yield_impl(result_type) override
         {
             // stackless coroutines don't support suspension
             HPX_ASSERT(false);

--- a/libs/coroutines/include/hpx/coroutines/stackless_coroutine.hpp
+++ b/libs/coroutines/include/hpx/coroutines/stackless_coroutine.hpp
@@ -58,7 +58,7 @@ namespace hpx { namespace threads { namespace coroutines {
             util::unique_function_nonser<result_type(arg_type)>;
 
         stackless_coroutine(functor_type&& f, thread_id_type id,
-            std::ptrdiff_t stack_size = default_stack_size)
+            std::ptrdiff_t /*stack_size*/ = default_stack_size)
           : f_(std::move(f))
           , state_(ctx_ready)
           , id_(id)

--- a/libs/datastructures/include/hpx/datastructures/tuple.hpp
+++ b/libs/datastructures/include/hpx/datastructures/tuple.hpp
@@ -886,7 +886,7 @@ namespace hpx { namespace util {
         {
             template <typename TTuple, typename UTuple>
             static constexpr HPX_HOST_DEVICE HPX_FORCEINLINE bool call(
-                TTuple const& t, UTuple const& u)
+                TTuple const&, UTuple const&)
             {
                 return true;
             }

--- a/libs/execution/include/hpx/execution/detail/post_policy_dispatch.hpp
+++ b/libs/execution/include/hpx/execution/detail/post_policy_dispatch.hpp
@@ -24,7 +24,7 @@ namespace hpx { namespace parallel { namespace execution { namespace detail {
     struct post_policy_dispatch
     {
         template <typename F, typename... Ts>
-        static void call(Policy const& policy,
+        static void call(Policy const&,
             hpx::util::thread_description const& desc,
             threads::thread_pool_base* pool, threads::thread_priority priority,
             threads::thread_stacksize stacksize,
@@ -38,7 +38,7 @@ namespace hpx { namespace parallel { namespace execution { namespace detail {
         }
 
         template <typename F, typename... Ts>
-        static void call(Policy const& policy,
+        static void call(Policy const&,
             hpx::util::thread_description const& desc,
             threads::thread_priority priority,
             threads::thread_stacksize stacksize,
@@ -68,7 +68,7 @@ namespace hpx { namespace parallel { namespace execution { namespace detail {
     struct post_policy_dispatch<launch::fork_policy>
     {
         template <typename F, typename... Ts>
-        static void call(launch::fork_policy const& policy,
+        static void call(launch::fork_policy const&,
             hpx::util::thread_description const& desc,
             threads::thread_pool_base* pool, threads::thread_priority priority,
             threads::thread_stacksize stacksize,

--- a/libs/execution/include/hpx/execution/executors/execution_parameters.hpp
+++ b/libs/execution/include/hpx/execution/executors/execution_parameters.hpp
@@ -134,7 +134,7 @@ namespace hpx { namespace parallel { namespace execution {
             template <typename AnyParameters, typename Executor>
             HPX_FORCEINLINE static std::size_t call_param(
                 hpx::traits::detail::wrap_int, AnyParameters&&, Executor&&,
-                std::size_t cores, std::size_t num_tasks)
+                std::size_t, std::size_t)
             {
                 // return zero chunks which will tell the implementation to
                 // calculate the number of chunks either based on a
@@ -211,8 +211,7 @@ namespace hpx { namespace parallel { namespace execution {
             // Check whether the parameter object implements this function
             template <typename AnyParameters, typename Executor>
             HPX_FORCEINLINE static void call_param(
-                hpx::traits::detail::wrap_int, AnyParameters&& params,
-                Executor&& exec)
+                hpx::traits::detail::wrap_int, AnyParameters&&, Executor&&)
             {
             }
 
@@ -278,8 +277,7 @@ namespace hpx { namespace parallel { namespace execution {
             // Check whether the parameter object implements this function
             template <typename AnyParameters, typename Executor>
             HPX_FORCEINLINE static std::size_t call_param(
-                hpx::traits::detail::wrap_int, AnyParameters&& params,
-                Executor&& exec)
+                hpx::traits::detail::wrap_int, AnyParameters&&, Executor&&)
             {
                 return get_os_thread_count();
             }

--- a/libs/futures/include/hpx/futures/packaged_continuation.hpp
+++ b/libs/futures/include/hpx/futures/packaged_continuation.hpp
@@ -480,7 +480,7 @@ namespace hpx { namespace lcos { namespace detail {
         template <typename Spawner, typename Policy>
         void attach(Future const& future,
             typename std::remove_reference<Spawner>::type& spawner,
-            Policy&& policy, error_code& ec = throws)
+            Policy&& policy, error_code& /*ec*/ = throws)
         {
             typedef typename traits::detail::shared_state_ptr_for<Future>::type
                 shared_state_ptr;
@@ -516,7 +516,7 @@ namespace hpx { namespace lcos { namespace detail {
         template <typename Spawner, typename Policy>
         void attach(Future const& future,
             typename std::remove_reference<Spawner>::type&& spawner,
-            Policy&& policy, error_code& ec = throws)
+            Policy&& policy, error_code& /*ec*/ = throws)
         {
             typedef typename traits::detail::shared_state_ptr_for<Future>::type
                 shared_state_ptr;
@@ -553,7 +553,7 @@ namespace hpx { namespace lcos { namespace detail {
         template <typename Spawner, typename Policy>
         void attach_nounwrap(Future const& future,
             typename std::remove_reference<Spawner>::type& spawner,
-            Policy&& policy, error_code& ec = throws)
+            Policy&& policy, error_code& /*ec*/ = throws)
         {
             typedef typename traits::detail::shared_state_ptr_for<Future>::type
                 shared_state_ptr;
@@ -589,7 +589,7 @@ namespace hpx { namespace lcos { namespace detail {
         template <typename Spawner, typename Policy>
         void attach_nounwrap(Future const& future,
             typename std::remove_reference<Spawner>::type&& spawner,
-            Policy&& policy, error_code& ec = throws)
+            Policy&& policy, error_code& /*ec*/ = throws)
         {
             typedef typename traits::detail::shared_state_ptr_for<Future>::type
                 shared_state_ptr;

--- a/libs/init_runtime/include/hpx/hpx_init_impl.hpp
+++ b/libs/init_runtime/include/hpx/hpx_init_impl.hpp
@@ -118,7 +118,7 @@ namespace hpx {
     /// function. It will set up the HPX runtime environment and schedule the
     /// function given by \p f as a HPX thread.
     inline int init(
-        std::nullptr_t f, int argc, char** argv, init_params const& params)
+        std::nullptr_t, int argc, char** argv, init_params const& params)
     {
         util::function_nonser<int(hpx::program_options::variables_map&)> main_f;
         if (argc == 0 || argv == nullptr)
@@ -411,7 +411,7 @@ namespace hpx {
     }
 
     inline int init(
-        std::nullptr_t f, int argc, char** argv, hpx::runtime_mode mode)
+        std::nullptr_t, int argc, char** argv, hpx::runtime_mode mode)
     {
         HPX_ASSERT(argc != 0 && argv != nullptr);
 

--- a/libs/init_runtime/include/hpx/hpx_start_impl.hpp
+++ b/libs/init_runtime/include/hpx/hpx_start_impl.hpp
@@ -126,7 +126,7 @@ namespace hpx {
     /// immediately after that. Use `hpx::wait` and `hpx::stop` to synchronize
     /// with the runtime system's execution.
     inline bool start(
-        std::nullptr_t f, int argc, char** argv, init_params const& params)
+        std::nullptr_t, int argc, char** argv, init_params const& params)
     {
         util::function_nonser<int(hpx::program_options::variables_map&)> main_f;
         if (argc == 0 || argv == nullptr)

--- a/libs/io_service/include/hpx/io_service/io_service_thread_pool.hpp
+++ b/libs/io_service/include/hpx/io_service/io_service_thread_pool.hpp
@@ -30,7 +30,7 @@ namespace hpx { namespace threads { namespace detail {
         explicit io_service_thread_pool(
             threads::thread_pool_init_parameters const& init);
 
-        void print_pool(std::ostream& os) {}
+        void print_pool(std::ostream&) {}
 
         ///////////////////////////////////////////////////////////////////////
         hpx::state get_state() const;

--- a/libs/serialization/include/hpx/serialization/serialize_buffer.hpp
+++ b/libs/serialization/include/hpx/serialization/serialize_buffer.hpp
@@ -289,7 +289,7 @@ namespace hpx { namespace serialization {
 
         ///////////////////////////////////////////////////////////////////////
         template <typename Archive>
-        void save(Archive& ar, unsigned int const version) const
+        void save(Archive& ar, unsigned int const) const
         {
             ar << size_ << alloc_;
             // -V128
@@ -302,7 +302,7 @@ namespace hpx { namespace serialization {
 
         ///////////////////////////////////////////////////////////////////////
         template <typename Archive>
-        void load(Archive& ar, unsigned int const version)
+        void load(Archive& ar, unsigned int const)
         {
             ar >> size_ >> alloc_;
             // -V128

--- a/libs/thread_support/include/hpx/thread_support/set_thread_name.hpp
+++ b/libs/thread_support/include/hpx/thread_support/set_thread_name.hpp
@@ -12,13 +12,14 @@
 #include <windows.h>
 
 namespace hpx { namespace util {
-    void set_thread_name(char const* threadName, DWORD dwThreadID = DWORD(-1));
+    void set_thread_name(
+        char const* /*thread_name*/, DWORD /*thread_id*/ = DWORD(-1));
 }}    // namespace hpx::util
 
 #else
 
 namespace hpx { namespace util {
-    inline void set_thread_name(char const* threadName) {}
+    inline void set_thread_name(char const* /*thread_name*/) {}
 }}    // namespace hpx::util
 
 #endif

--- a/libs/threading_base/include/hpx/threading_base/external_timer.hpp
+++ b/libs/threading_base/include/hpx/threading_base/external_timer.hpp
@@ -299,25 +299,23 @@ namespace hpx { namespace util {
         {
         };
 
-        inline std::shared_ptr<task_wrapper> new_task(
-            thread_description const& description,
-            std::uint32_t parent_locality_id,
-            threads::thread_id_type const& parent_task)
+        inline std::shared_ptr<task_wrapper> new_task(thread_description const&,
+            std::uint32_t, threads::thread_id_type const&)
         {
             return nullptr;
         }
 
         inline std::shared_ptr<task_wrapper> update_task(
-            std::shared_ptr<task_wrapper> wrapper,
-            thread_description const& description)
+            std::shared_ptr<task_wrapper>, thread_description const&)
         {
             return nullptr;
         }
 
         struct scoped_timer
         {
-            explicit scoped_timer(std::shared_ptr<task_wrapper> data_ptr) {}
-            ~scoped_timer() {}
+            explicit scoped_timer(std::shared_ptr<task_wrapper>) {}
+            ~scoped_timer() = default;
+
             void stop(void) {}
             void yield(void) {}
         };

--- a/libs/threading_base/include/hpx/threading_base/register_thread.hpp
+++ b/libs/threading_base/include/hpx/threading_base/register_thread.hpp
@@ -225,7 +225,7 @@ namespace hpx { namespace threads {
     inline threads::thread_id_type register_thread_plain(
         threads::thread_pool_base* pool, threads::thread_init_data& data,
         threads::thread_state_enum initial_state = threads::pending,
-        bool run_now = true, error_code& ec = throws)
+        bool /*run_now*/ = true, error_code& ec = throws)
     {
         HPX_ASSERT(pool);
         threads::thread_id_type id = threads::invalid_thread_id;

--- a/libs/threading_base/include/hpx/threading_base/set_thread_state.hpp
+++ b/libs/threading_base/include/hpx/threading_base/set_thread_state.hpp
@@ -295,8 +295,8 @@ namespace hpx { namespace threads { namespace detail {
     /// This thread function is used by the at_timer thread below to trigger
     /// the required action.
     inline thread_result_type wake_timer_thread(thread_id_type const& thrd,
-        thread_state_enum newstate, thread_state_ex_enum newstate_ex,
-        thread_priority priority, thread_id_type const& timer_id,
+        thread_state_enum /*newstate*/, thread_state_ex_enum /*newstate_ex*/,
+        thread_priority /*priority*/, thread_id_type const& timer_id,
         std::shared_ptr<std::atomic<bool>> const& triggered,
         bool retry_on_active, thread_state_ex_enum my_statex)
     {

